docs/signature.md

```diff
@@ -0,0 +1,93 @@
+# Image Signature Specification
+
+**Version 0.1**
+
+## Introduction
+
+This document defines a detached container image signature object and signing methods.
+
+## Signature Format
+
+```js
+{
+    "critical": {/* required fields */
```

Comment:
This is a forward compatibility mechanism. Sure, a current implementation of a current spec does not need this explicit designation, but when the spec changes in a year or two (which is pretty likely), and new signatures start appearing, old implementations still need to be able to process them somehow. Separating the new fields between `critical` (if the old implementation doesn’t understand the data, it **must** reject the signature) and `optional` (if the old implementation doesn’t understand the data, it can proceed anyway) is a way for old implementations unaware of the updated spec to handle the signatures reasonably correctly.

And yes, this is a very common mechanism, e.g. https://tools.ietf.org/html/rfc4880#section-5.2.3.1
>  Bit 7 of the subpacket type is the "critical" bit.  If set, it denotes that the subpacket is one that is critical for the evaluator of the signature to recognize.  If a subpacket is encountered that is marked critical but is unknown to the evaluating software, the evaluator SHOULD consider the signature to be in error.

or https://tools.ietf.org/html/rfc5280#section-4.2
> Each extension in a certificate is designated as either critical or non-critical.  A certificate-using system MUST reject the certificate if it encounters   a critical extension it does not recognize or a critical extension   that contains information that it cannot process.